### ISSUE TITLE
Release process without protection bypass

### ISSUE THREAD

**Issue by Krinkle** [qunitjs/qunit#1350]
Reading through [RELEASE.md](https://github.com/qunitjs/qunit/blob/2.8.0/RELEASE.md) (Thanks @trentmwillis for writing this!), it noticed it says one has to temporarily disable branch protection for the `Build: Prepare @VERSION release`  commit.

This issue is to evaluate which parts of the release process would need to change to avoid this step, and whether would want to change it.

**Comment by Krinkle** [comment]
Note that the branch protection requirement for passing CI can be fulfilled even without the overhead of a pull request; by pushing the commit to a temporary branch (e.g. "release"), then wait for CI to complete. After that, it is allowed for the same commit to be pushed to master.

The branch protection requirement for 1 code review does require a pull request, naturally. If we want to continue do not require peer-review for releases, then perhaps we could consider lifting this requirement more generally; in favour of adopting a purely social policy (that is, we'd agree to still only land PRs after CR, with the exception of the release commit).

Doing that would have the benefit of still letting CI run on the release commit, and more importantly, it would mean releasing QUnit no longer requires changing repository settings (or repository Admin rights).

**Comment by trentmwillis** [comment]
I agree it would be beneficial to not need the setting changes when performing a release. Note however that currently to do a release, you just need to disable the restrictions for _administrators_. We could change this to _always be disabled for administrators_ and adopt a social policy as you mention to continue ensuring other PRs/commits from project members get reviewed and pass status checks before getting merged.
